src/evaluation/evaluator.h

```diff
@@ -6,322 +6,348 @@
 #include "evaluation/pv_table.h"
 #include "evaluation/repetition.h"
 #include "evaluation/static_evaluation.h"
+
 #include "fmt/ranges.h"
 #include "movegen/move_types.h"
 #include "syzygy/syzygy.h"
 #include <atomic>
 #include <engine/zobrist_hashing.h>
 
 #include <chrono>
+#include <condition_variable>
+#include <future>
 #include <mutex>
 #include <thread>
 
 namespace evaluation {
 
-class Evaluator {
+enum SearchType {
+    Default,
+    NullSearch,
+};
+
+enum SearchState {
+    Idle,
+    Searching,
+    Kill
+};
+
+struct MoveResult {
+    uint64_t hash;
+    BitBoard board;
+};
+
+struct SearcherResult {
+    int32_t score;
+    uint8_t searcherId;
+    movegen::Move pvMove;
+    uint8_t searchedDepth;
+};
+
+// Wrapper around a static array for intuitive iteration
+template<typename T, uint8_t maxSize>
+class StackVector {
```

Comment:
Not the biggest fan of custom containers unless it's something we find ourselves using on a regular basis. 
A simple pre-allocated array and an index can do the same :)